File: Javascript/LongestSubstringWithoutRepeatingCharacters.js
var lengthOfLongestSubstring = function(s){
    let left = 0
    let right = 0
    let max_len = 0
    let window = new Set()

    while(right<s.length){
        if(!(window.has(s[right]))){
            window.add(s[right])
            right += 1
            max_len = Math.max(max_len,right-left)
        }
        else{
            window.delete(s[left])
            left += 1
        }
    }

    return max_len
}

File: Python/LongestSubstringWithoutRepeatingCharacters.py
def lengthOfLongestSubstring(s):
    left,right,max_len = 0,0,0
    window = set()

    while right < len(s):
        if s[right] not in window:
            window.add(s[right])
            right += 1
            max_len = max(max_len,right-left)
        else:
            window.remove(s[left])
            left +=1 
    
    return max_len